Sources/Core/EthereumABI/ABIEncoding.swift

```diff
@@ -41,37 +40,35 @@ extension ABIEncoder {
         case let v as UInt64:
             return BigUInt(v)
         case let v as Int:
-            return BigUInt(v)
+            return v < 0 ? nil : BigUInt(v)
         case let v as Int8:
-            return BigUInt(v)
+            return v < 0 ? nil : BigUInt(v)
         case let v as Int16:
-            return BigUInt(v)
+            return v < 0 ? nil : BigUInt(v)
         case let v as Int32:
-            return BigUInt(v)
+            return v < 0 ? nil : BigUInt(v)
```

Comment:
Why are we casting any type to BigUInt rather keeping the original one?

Reply:
Because we have implementations that properly encode BigInt and BigUInt for the ABI (e.g. function calls).
We are using BigInt and BigUInt as the primary type for numbers across the library, as we may have very large numbers sometimes, so this is the reason.